<launch>
    <node name="gps_offset_minus_node" pkg="njust_pack" type="gps.py" output="screen">
        <param name="x_offset" value="233213.1585182654"/>
        <param name="y_offset" value="3696889.9050581"/>
        <remap from="GPS_raw_odom" to="/GPS/utm_odom"/>
        <remap from="GPS_odom" to="/GPS/final_odom"/>
    </node>

    <node name="utm_odometry_node" pkg="gps_common" type="utm_odometry_node">
        <remap from="fix" to="/GPS/fix"/>
        <remap from="odom" to="/GPS/utm_odom"/>
        <param name="frame_id" value="map"/>
        <param name="child_frame_id" value="GPS_odom"/>
    </node>

    <!-->
    <node name="navsat_transform_node" pkg="robot_localization" type="navsat_transform_node">
        <param name="zero_altitude" value="true"/>
        <remap from="imu/data" to="imu"/>
        <remap from="odometry/filtered" to="odometry/filtered"/>
        <remap from="gps/fix" to="GPS/fix"/>
    </node>
    <-->

    <node name="nmea_serial_driver_node" pkg="nmea_navsat_driver" type="nmea_serial_driver" output="screen" ns="GPS">
        <!-- <param name="port" value="/dev/njustWRGPS"/>  -->
        <param name="port" value="/dev/ttyUSB0"/>
        <param name="baud" value="115200" />
        <param name="frame_id" value="GPS" />
        <param name="time_ref_source" value="GPS" />
        <param name="useRMC" value="false" />
    </node>



    <node name="ekf_localization_node" pkg="robot_localization" type="ekf_localization_node">
        <param name="two_d_mode" value="true"/>
        <param name="frequency" value="50"/>

        <param name="imu0" value="/imu"/>
        <rosparam param="imu0_config">[false,  false,  false,
                                false, false, false,
                                false,  false, false,
                                true, true, true,
                                true, true, true]</rosparam>
        <param name="odom0" value="/odom"/>
        <rosparam param="odom0_config">[false,  false,  false,
                                false, false, false,
                                true,  true, true,
                                false, false, false,
                                false, false, false]</rosparam>
        <param name="odom1" value="/GPS/final_odom"/>
        <rosparam param="odom1_config">[true,  true,  true,
                                false, false, false,
                                false,  false, false,
                                false, false, false,
                                false, false, false]</rosparam>
    </node>


    <node pkg="tf" type="static_transform_publisher" name="base_link_imu_tf_broadcaster" args="0 0 0.2 0 0 0 base_link imu 100" />
    <node pkg="tf" type="static_transform_publisher" name="base_link_gps_tf_broadcaster" args="0 0 0.2 0 0 0 base_link GPS 100" />
    <node name="test_node" pkg="njust_pack" type="test.py" output="screen">
    </node>

</launch>
